src/descriptor/policy.rs

```diff
@@ -76,6 +76,19 @@ pub enum PkOrF {
     XOnlyPubkey(XOnlyPublicKey),
     /// An extended key fingerprint
     Fingerprint(Fingerprint),
+    /// serialization for xpub origin metadata
+    XpubOrigin(XpubOrigin),
+}
+
+/// Raw public key or extended key fingerprint
+#[derive(Debug, Clone, Default, Serialize, PartialEq, Eq, Hash)]
+pub struct XpubOrigin {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    fingerprint: Option<Fingerprint>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    derivation_path: Option<DerivationPath>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    xkey: Option<ExtendedPubKey>,
```

Comment:
This seems to me representing a full XPub, not just the origin.. So we can name it better as just an `XPub`?

Reply:
Are you referring to the struct `XpubOrigin` or this field, `xkey`? 

In the contexts I'm most familiar working in, an xpub origin has more information than the xpub (extended public key) itself. Namely the origin is the xpub and the information describing how it was derived:  fingerprint for the root (`xfp` which might be a better name for that field) and the derivation path from root. The xpub only has its parent fingerprint and depth encoded (with some other data too of course.)  

As for renaming `xkey`, I forget why I put it as this, but I feel like I remember there being a reason, maybe some sort of naming clash. If that isn't a problem though then I'm not against renaming that to `xpub`.

Reply:
I was actually referring to the struct name `XpubOrigin`.. The name suggests that its an `origin` for `Xpub`, but its more like an `XpubWithOrigin`. But I agree its mostly nomenclature and there's no *right* way to do it.. I think just the doc comment as something like  `/// Extended Publickey with origin information` is good enough to clarify what this struct is for.. The struct name can remain as it is..

Reply:
OTOH if I am not wrong, after parsing a descriptor we will always have the origin information? If no fingerprint and path is provided in the descriptor, bdk assumes it as the master key? If that's the case we can simply name this struct `Xpub` and it will imply that origin information is already included in it.. As we never put an Xpub into a policy without its origin..

Reply:
Even if bdk never puts an xpub w/o origin details, if we're given one to interpret can you safely assume it's the root key? You don't need the origin information for watch-only wallets for example since you only need those details to tell the master key how to sign. We do something like this for unchained wallet configurations (which I want to use this tooling to convert to descriptors) where we only show actual origin information for keys that are yours (the clients) and blank out the origin info for those you don't have permissions to sign with in our system (for example the unchained key). We do this to preserve privacy and avoid data leaks. The xpub's relation to the master seed (origin information) should be of no concern to the other party that's signing if that key is not going to be signed with. So, in this context the origin information is optional and this would be true for all keys if you only want a watch-only descriptor.

Reply:
Yes I agree.. For watch-only we don't need the origin info and its not prudent to assume its a master key.. ACK on that..